### PR TITLE
Implement login and logout, closes #9

### LINKED ISSUE #9: As a user, I want to login to circuit, so that later I have access
@Iotka and @nrigheriu are responsible for this feature.
Also includes logout.

### PR DESCRIPTION
**Pull request by metalmatze**
This uses the build in functionality of django to authenticate users.

### CODE REVIEW

File: users/templates/users/user_login.html [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+{% block content %}
+    <div class="row">
+
+        <div class="col-sm-8 col-sm-offset-2 col-md-6 col-md-offset-3 col-lg-4 col-lg-offset-4">
+
+            <h2>Login</h2>
+
+            {% if messages %}
+                {% for message in messages %}
```
Comment (Gjum) [inline users/templates/users/user_login.html:11]: Is there no block or something for these dismissable alerts? Seems like this will lead to code duplication when used consistently.

Anyways, nice way to notify the user of any errors. Should be used more often.

File: users/views.py [hunk 2]
```diff
@@ -56,10 +59,39 @@ def UserCreateConfirmView(request, token):
         return redirect('users:signup')
```
Comment (Gjum) [inline users/views.py:58]: The user should be redirected to `users:login` instead of `users:signup` in line 57, i.e. when clicking on the mailed confirmation link.
(Can't comment on line 57 - outside the diff :P)

Comment (Gjum) [inline users/views.py:58]: @MetalMatze hotfix pls

File: users/views.py [hunk 3]
```diff
@@ -56,10 +59,39 @@ def UserCreateConfirmView(request, token):
...
+        password = request.POST['password']
+
+        user = authenticate(email=email, password=password)
+
+        if user is not None:
+            if user.is_active:
+                login(request, user)
+                return redirect('users:profile', pk=user.pk)
+            else:
+                messages.error(request, 'Your account is inactive.')
```
Comment (Gjum) [inline users/views.py]: Maybe a message that the user should click the mailed confirmation link would be useful.

File: users/views.py (hunk 2 above)
Comment (metalmatze) [inline users/views.py:58]: You're right. There are a few more things that need to be updated, this is just one. But I will fix this one.

File: users/templates/users/user_login.html (hunk 1 above)
Comment (metalmatze) [inline users/templates/users/user_login.html:11]: What do you mean by block?

Comment (Gjum) [inline users/templates/users/user_login.html:11]: The same template code would have to be written over and over again for every single form. So I thought of creating a template block for this. `include`ing another HTML might be another way.

**Comment by Gjum** [comment]
Ready for merge in my opinion. @MetalMatze ?

File: users/views.py (hunk 2 above)
Comment (Gjum) [inline users/views.py:58]: I'd volunteer to look around and do some updating in a separate PR after this is merged.
E.g. adding `user.is_authenticated` to templates and fixing links.

File: users/views.py [hunk 4]
```diff
@@ -51,15 +53,44 @@ def UserCreateConfirmView(request, token):
...
+                messages.error(request, 'Your account is inactive. Be sure to check your emails.')
+                return redirect('users:login')
+        else:
+            messages.error(request, message='Bad credentials.')
+            return redirect('users:login')
+
+
+def UserLogoutView(request):
+    logout(request)
+    return redirect('landingpage')
```
Comment (Gjum) [inline users/views.py:86]: Should we really always redirect to the landing page instead of just refreshing the current page when possible?
E.g. you view your profile, click `Logout`, then you still see your profile, but can't edit it for example.
